fail2ban/server/filter.py

```diff
@@ -898,22 +898,18 @@ def textToIp(text, useDns):
 		return ipList
 
 	@staticmethod
-	def cidr(i, n):
-		""" Convert an IP address string with a CIDR mask into a 32-bit
-			integer.
+	def addr2bin(ipstring, cidr=None):
+		""" Convert a string IPv4 address into decimal form.
+		If cidr is supplied, return the network address for the given block
 		"""
-		# 32-bit IPv4 address mask
-		MASK = 0xFFFFFFFFL
-		return ~(MASK >> n) & MASK & DNSUtils.addr2bin(i)
-
-	@staticmethod
-	def addr2bin(string):
-		""" Convert a string IPv4 address into an unsigned integer.
-		"""
-		return struct.unpack("!L", socket.inet_aton(string))[0]
+		if cidr is None:
+			return struct.unpack("!L", socket.inet_aton(ipstring))[0]
```

Comment:
here is another prove that it has nothing to do with 'dec':

```
*In [10]: '%x' % struct.unpack('!L', socket.inet_aton('10.1.2.3'))[0]
Out[10]: 'a010203'
```

as you can see -- it just packs those 4 numbers into 4 bytes.  I really see no relation to 'dec' here.